main.rb

```diff
@@ -0,0 +1,168 @@
+        yield item[i], i
+      elsif parm.arity == 1
+        yield(assoc item.keys[i])
+      else
+        yield item.keys[i], item[item.keys[i]], i
+      end
+      i += 1
+    end
+    item
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    items = self
+
+    items = items.to_a if items.is_a? Range
+
+    if items.is_a?(Array)
+
+      new_values = []
+
+      items.my_each { |i| new_values.push(i) if yield(i) == true }
+
+      new_values
+
+    elsif items.is_a?(Hash)
+
+      new_values = {}
+
+      items.my_each { |a, b| new_values.store(a, b) if yield(a, b) == true }
+      new_values
+    else
+      items
+    end
+  end
+
+  def my_all?(parms = nil)
+    item = self
+    results = false
+    if parms.nil? && !block_given?
+      item.my_each do |value|
+        results = value.nil? || !value == true
+        break if results
+      end
+      !results
+    elsif parms.nil?
+      item.my_each do |value|
+        results = yield(value)
+        break unless results
+      end
+      results
+    elsif parms.is_a?(Regexp)
+      item.my_each do |value|
+        results = value.to_s.match?(parms)
+        break unless results
+      end
+      results
+    elsif parms.is_a?(Class)
+      item.my_each do |value|
+        results = value.is_a?(parms)
+        break unless results
+      end
+      results
+    else
+      item.my_each do |value|
+        results = value == parms
+        break unless results
+      end
+      results
+    end
+  end
+
+  def my_any?(parms = nil)
+    my_each do |item|
+      if !parms.nil?
+        if parms.is_a?(Regexp)
+          return true if item.match(parms)
+        elsif parms.is_a?(Class)
+          return true if item.is_a?(parms)
+        elsif item == parms
+          return true
+        end
+      elsif block_given?
+        return true if yield(item)
+      elsif item
+        return true
+      end
+    end
+    false
+  end
+
+  def my_none?(parms = nil, &block)
+    !my_any?(parms, &block)
+  end
+
+  def my_count(*parms)
+    return_value = 0
+    sum_one = ->(bool) { return_value += 1 if bool }
+
+    if block_given?
+      my_each { |i| sum_one.call(yield(i)) }
+    else
+      my_each { |i| parms.empty? ? return_value += 1 : sum_one.call(parms[0] == i) }
+    end
+
+    return_value
+  end
+
+  def my_map(parms = nil)
+    return to_enum(:my_map) unless block_given?
+
+    mapped = []
+    my_each { |i| mapped << parms.call(i) } if block_given? && parms
+    my_each { |i| mapped << yield(i) } if parms.nil?
+    mapped
+  end
+
+  def my_inject(results = nil, sym = nil, &parms)
+    results = results.to_sym if results.is_a?(String) && !sym && !parms
+
+    if results.is_a?(Symbol) && !sym
+      parms = results.to_proc
+      results = nil
+    end
+
+    sym = sym.to_sym if sym.is_a?(String)
+    parms = sym.to_proc if sym.is_a?(Symbol)
+
+    my_each { |i| results = results.nil? ? i : parms.yield(results, i) }
+    results
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78313563-5db28e00-7514-11ea-8011-19c350c14f73.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
(1..50).my_inject(:+) == (1..50).inject(:+) => 13
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78313646-b2ee9f80-7514-11ea-932f-e862d9a79b8f.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)